app/web/features/auth/notifications/NotificationSettings.tsx

```diff
@@ -0,0 +1,58 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import CircularProgress from "components/CircularProgress";
+import { notificationSettingsQueryKey } from "features/queryKeys";
+import { Error as GrpcError } from "grpc-web";
+import { GetNotificationSettingsRes } from "proto/notifications_pb";
+import { useTranslation } from "react-i18next";
+import { useQuery, useQueryClient } from "react-query";
+import { service } from "service";
+
+type NotificationSettingsProps = {
+  className?: string;
+};
+
+export default function NotificationSettings(props: NotificationSettingsProps) {
+  const { t } = useTranslation("auth");
+
+  const queryClient = useQueryClient();
+
+  const { data, error, isLoading } = useQuery<
+    GetNotificationSettingsRes.AsObject,
+    GrpcError
+  >(
+    notificationSettingsQueryKey,
+    service.notifications.getNotificationSettings
+  );
+
+  const toggleNewNotifications = async () => {
+    await service.notifications.setNotificationSettings(
+      !data?.newNotificationsEnabled
+    );
+    queryClient.invalidateQueries(notificationSettingsQueryKey);
+  };
+
+  const { className } = props;
```

Comment:
We usually destructure directly in the function definition.

Reply:
OK, it's 'cos I'm a grummy copy-paste code... :S